orc8r/cloud/easy_deploy/docker/docker_root/scripts/configlib.py

```diff
@@ -0,0 +1,152 @@
+
+        with open(config_dir + cfg_fn, 'w') as outfile:
+            json.dump(cfgs, outfile)
+
+        # additionally configure AWS if necessary
+        self.configure_aws(config_type, cfgs)
+
+    def check_config(self, config_dir: str, config_type: str) -> bool:
+        ''' check if all mandatory options of a specific config_type is set '''
+        print("\nChecking %s deployment variables\n" % config_type)
+        cfgs = {}
+        try:
+            deploy_cfg_fn =  tf_vars_filename(config_type)
+            with open(config_dir + deploy_cfg_fn) as f:
+                cfgs = json.load(f)
+        except OSError:
+            pass
+
+        valid = True
+        for k, v in self.vars[config_type].items():
+            if not v['Required']:
+                continue
+            v = cfgs.get(k)
+            if not v:
+                print("%s not configured" % k)
+                valid = False
+
+        return valid
+
+    def info(self, config_dir: str, config_type: str):
+        ''' pretty print vars yml '''
+        print ("%s Configuration Options" % config_type)
+        x = PrettyTable()
+        x.field_names = ["Name", "Description", "Type", "Required", "Component"]
+        for k, v in self.vars[config_type].items():
+            x.add_row([k, v["Description"], v["Type"], v["Required"], v["Component"]])
+        return print(x)
+
+    def show(self, config_dir: str, config_type: str):
+        ''' pretty print existing configuration '''
+        print ("%s Configuration" % config_type)
+        deployment_cfgs = {}
+        try:
+            deploy_cfg_fn =  tf_vars_filename(config_type)
+            with open(config_dir + deploy_cfg_fn) as f:
+                deployment_cfgs = json.load(f)
+        except OSError:
+            pass
+        x = PrettyTable()
+        x.field_names = ["Name", "Configuration"]
+        for k, v in deployment_cfgs.items():
+            x.add_row([k, v])
+        return print(x)
+
+    def handle_args(self, config_dir: str, config_type: str, args):
+        if self.vars.get(config_type):
+            if args.info:
+                return self.info(config_dir, config_type)
+            elif args.show:
+                return self.show(config_dir, config_type)
+            elif args.check:
+                if (not self.check_config(config_dir, config_type)):
+                    sys.exit(1)
+            else:
+                self.configure(config_dir, config_type)
+
+    def __call__(self, config_dir: str, args):
+        inp_config_type = get_config_type_args(args)
+        for config_type in (INFRA_CONFIG_TYPE,
+                            PLATFORM_CONFIG_TYPE,
+                            SERVICE_CONFIG_TYPE):
+            if inp_config_type != None and config_type != inp_config_type:
+                continue
+            self.handle_args(config_dir, config_type, args)
+
+configure_deployment = None
+def configure_all(constants: dict, args):
+    global configure_deployment
```

Comment:
These are python functors. This is instantiating the configure_deployment function. I made this a functor so that I can hold the state for vars parsed during the initialization. Once the vars is initialized, we just use it as a regular function. Looking at this again, I feel it is introducing unnecessary confusion. I will clean it up.